File: frontend/src/types/product.ts
export interface Product {
  productId: number
  shopId: number
  categoryId: number
  productName: string
  productDescription: string | null
  productPrice: number
  isVisible: boolean
  createdAt: string
  images: ProductImage[]
  options: ProductOption[]
}

export interface ProductOption {
  optionId: number
  productId: number
  color: string | null
  size: string | null
  stockQty: number
  alertThreshold: number | null
}

export interface ProductImage {
  imageId: number
  productId: number
  imageUrl: string | null
  isMain: boolean
  sortOrder: number | null
}

export interface Category {
  categoryId: number
  parentId: number | null
  categoryName: string | null
  depth: number | null
  sortOrder: number
  children?: Category[]
}

File: backend/src/main/java/com/odersite/domain/product/entity/Product.java
package com.odersite.domain.product.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "PRODUCT")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "product_id")
    private Integer productId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "shop_id", nullable = false)
    private Shop shop;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id", nullable = false)
    private Category category;

    @Column(name = "product_name", nullable = false, length = 200)
    private String productName;

    @Column(name = "product_description", columnDefinition = "TEXT")
    private String productDescription;

    @Column(name = "product_price", nullable = false)
    private Integer productPrice;

    @Column(name = "is_visible")
    @Builder.Default
    private Boolean isVisible = true;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<ProductOption> options = new ArrayList<>();

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<ProductImage> images = new ArrayList<>();

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
    }

    public void update(String productName, String productDescription, Integer productPrice, Category category) {
        this.productName = productName;
        this.productDescription = productDescription;
        this.productPrice = productPrice;
        this.category = category;
    }

    public void toggleVisibility(boolean visible) {
        this.isVisible = visible;
    }
}
